projects/pt1/python/torch_mlir/jit_ir_importer/build_tools/abstract_interp_lib_gen.py

```diff
@@ -1501,6 +1501,46 @@ def aten〇addcmul〡shape(self: List[int], tensor1: List[int], tensor2: List[in
 def aten〇addcdiv〡shape(self: List[int], tensor1: List[int], tensor2: List[int], value: float = 1) -> List[int]:
     return upstream_shape_functions.broadcast(self, upstream_shape_functions.broadcast(tensor1, tensor2))
 
+@check_shape_function([
+    Invocation(TensorOfShape(1,5,5), [5,5], [1,5], [1,1], [0,0], [1,1]), # basic case
+    Invocation(TensorOfShape(1,4,5), [6,6], [2,2], [1,5], [0,0], [1,1]), # dilation
+    Invocation(TensorOfShape(1,5,15), [5,5], [1,5], [1,1], [0,1], [1,1]), # padding
+    Invocation(TensorOfShape(1,9,4), [5,5], [3,3], [1,1], [0,0], [2,2]), # stride
+    ErrorInvocation(TensorOfShape(1,5,5), [5,5], [1,7], [1,1], [0,0], [1,1]), # mismatch of sliding blocks
+])
+def aten〇col2im〡shape(self: List[int], output_size: List[int], kernel_size: List[int], dilation: List[int], padding: List[int], stride: List[int]) -> List[int]:
+    assert len(output_size) == 2, "output_size is expected to have length 2"
+    assert len(kernel_size) == 2, "kernel_size is expected to have length 2"
+    assert len(dilation) == 2, "dilation is expected to have length 2"
+    assert len(stride) == 2, "stride is expected to have length 2"
+    assert len(padding) == 2, "padding is expected to have length 2"
+
+    assert kernel_size[0] > 0 and kernel_size[1] > 0, "kernel size should be greater than 0"
+    assert dilation[0] > 0 and dilation[1] > 0, "dilation should be greater than 0"
+    assert padding[0] >= 0 and padding[1] >= 0, "padding should be non negative"
+    assert stride[0] > 0 and stride[1] > 0, "stride must be greater than 0"
+
+    ndim = len(self)
+    assert (ndim == 2 and self[0] != 0 and self[1] != 0) or (ndim == 3 and self[1] != 0 and self[2] != 0), "Expected 2D or 3D (batch mode) tensor for input with possibly 0 batch size and non zero dimensions for input"
```

Comment:
I think this check should be at the top since it checks for the supported input size.

Reply:
Fixed!